go.mod

```
@@ -31,5 +32,6 @@ require (
 	google.golang.org/api v0.32.0
 	gopkg.in/yaml.v2 v2.3.0
 	k8s.io/utils v0.0.0-20200912215256-4140de9c8800
+	sigs.k8s.io/k8s-container-image-promoter v1.0.7
```

```suggestion
	sigs.k8s.io/k8s-container-image-promoter v2.4.0
```

I've just cut a new tag of k-sigs/cip for you to use --> https://github.com/kubernetes-sigs/k8s-container-image-promoter/releases/tag/v2.4.0
cc: @listx

---

I keep getting this error:
`require sigs.k8s.io/k8s-container-image-promoter: version "v2.4.0" invalid: module contains a go.mod file, so major version must be compatible: should be v0 or v1, not v2`

---

We're running into the dreaded v2 semantic import versioning problem:

ref: https://blog.golang.org/v2-go-modules, https://donatstudios.com/Go-v2-Modules, https://peter.bourgon.org/blog/2020/09/14/siv-is-unsound.html

I've opened https://github.com/kubernetes-sigs/k8s-container-image-promoter/pull/261 to discuss creating submodules for the container image promoter to get around this limitation.